tests/integration/targets/instance_config_vpc/tasks/main.yaml

```diff
@@ -0,0 +1,174 @@
+        configs:
+          - label: cool-config
+            devices:
+              sda:
+                disk_label: test-disk
+            interfaces:
+              - purpose: public
+                primary: true
+              - purpose: vpc
+                subnet_id: '{{ create_subnet.subnet.id }}'
+                ipv4:
+                  vpc: 10.0.0.4
+                ip_ranges: [ "10.0.0.7/32" ]
+        wait: false
+        booted: false
+        state: present
+      register: update_instance
+
+    - name: Assert instance updated
+      assert:
+        that:
+          - update_instance.changed
+
+          - update_instance.configs[0].interfaces[0].purpose == 'public'
+          - update_instance.configs[0].interfaces[0].primary
+
+          - update_instance.configs[0].interfaces[1].purpose == 'vpc'
+          - update_instance.configs[0].interfaces[1].subnet_id == create_subnet.subnet.id
+          - update_instance.configs[0].interfaces[1].vpc_id == create_vpc.vpc.id
+          - update_instance.configs[0].interfaces[1].ip_ranges[0] == '10.0.0.7/32'
+          - update_instance.configs[0].interfaces[1].ipv4.nat_1_1 == ""
+          - update_instance.configs[0].interfaces[1].ipv4.vpc == '10.0.0.4'
+
+    - name: Don't change the config interfaces
+      linode.cloud.instance:
+        label: 'ansible-test-{{ r }}'
+        region: us-mia
+        type: g6-standard-1
+        disks:
+          - label: test-disk
+            filesystem: ext4
+            size: 10
+        configs:
+          - label: cool-config
+            devices:
+              sda:
+                disk_label: test-disk
+            interfaces:
+              - purpose: public
+                primary: true
+              - purpose: vpc
+                subnet_id: '{{ create_subnet.subnet.id }}'
+                ipv4:
+                  vpc: 10.0.0.4
+                ip_ranges: [ "10.0.0.7/32" ]
+        wait: false
+        booted: false
+        state: present
+      register: unchanged_instance
+
+    - name: Assert unchanged
+      assert:
+        that:
+          - unchanged_instance.changed == False
+
+  always:
+    - ignore_errors: yes
+      block:
+        - name: Delete a Linode instance
+          linode.cloud.instance:
+            label: '{{ create_instance.instance.label }}'
+            state: absent
+          register: delete_instance
+
+        - name: Assert instance delete succeeded
+          assert:
+            that:
+              - delete_instance.changed
+              - delete_instance.instance.id == create_instance.instance.id
+
+        - name: Delete the VPC
+          linode.cloud.vpc:
+            label: 'ansible-test-{{ r }}'
+            state: absent
+          register: delete_vpc
+
+  environment:
+    LINODE_UA_PREFIX: '{{ ua_prefix }}'
+    LINODE_API_TOKEN: '{{ api_token }}'
+    LINODE_API_URL: '{{ api_url }}'
+    LINODE_API_VERSION: '{{ api_version }}'
+    REQUESTS_CA_BUNDLE: '{{ ca_file or "" }}'
```

Comment:
We've since added support for the `LINODE_CA` environment variable so I think we can replace this in all of the test files in this PR :+1:

Reply:
Sounds good!